File: 2019-8-6/AVFoundation/AVFoundation/ViewController.m
//
//  ViewController.m
//  AVFoundation
//
//  Created by 吉腾蛟 on 2019/8/6.
//  Copyright © 2019 JY. All rights reserved.
//

#import "ViewController.h"
#import <MobileCoreServices/MobileCoreServices.h>
#import <AVFoundation/AVFoundation.h>
#import <AVKit/AVKit.h>


@interface ViewController ()<UIImagePickerControllerDelegate,UINavigationControllerDelegate>
@property (weak, nonatomic) IBOutlet UIImageView *imageView;
@property(nonatomic,strong)UIImagePickerController *imagePickerCon;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

-(UIImagePickerController *)imagePickerCon{
    if (!_imagePickerCon) {
        _imagePickerCon=[[UIImagePickerController alloc] init];
        
        //采集源类型
        _imagePickerCon.sourceType=UIImagePickerControllerSourceTypePhotoLibrary;
        
        //媒体类型
        _imagePickerCon.mediaTypes=[NSArray arrayWithObject:(__bridge NSString *)kUTTypeImage];
        
        //设置代理
        _imagePickerCon.delegate=self;
    }
    return _imagePickerCon;
}

- (IBAction)selectImage:(id)sender {
    //通过摄像头来采集
    if ([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera]) {
        self.imagePickerCon.sourceType=UIImagePickerControllerSourceTypeCamera;
    }else{
        //通过图片库来采集
        self.imagePickerCon.sourceType=UIImagePickerControllerSourceTypePhotoLibrary;
    }
    
    [self presentViewController:self.imagePickerCon animated:YES completion:nil];
}
//完成采集图片后的回调处理
-(void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary<UIImagePickerControllerInfoKey,id> *)info{
    //获取媒体类型
    NSString *type=info[UIImagePickerControllerMediaType];
    //如果媒体类型是图片类型
    if ([type isEqualToString:(__bridge NSString *)kUTTypeImage]) {
        //获取采集到的图片
        UIImage *image=info[UIImagePickerControllerOriginalImage];
        //显示到UI界面
        self.imageView.image=image;
    }
    
    [self dismissViewControllerAnimated:YES completion:nil];
}
//取消采集图片的处理
-(void)imagePickerControllerDidCancel:(UIImagePickerController *)picker{
    NSLog(@"");
    
    [self dismissViewControllerAnimated:YES completion:nil];
}


@end

File: 2019-5-29(SWift)/QuartzFun/QuartzFun/ViewController.swift
//
//  ViewController.swift
//  QuartzFun
//
//  Created by 吉腾蛟 on 2019/9/25.
//  Copyright © 2019 JY. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var colorControl: UISegmentedControl!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
    }

    @IBAction func changeColor(_ sender: UISegmentedControl) {
        let drawingColorSelection = DrawingColor(rawValue: UInt(sender.selectedSegmentIndex))
        if let drawingColor = drawingColorSelection {
            let funView = view as! QuartzFunView
            switch drawingColor {
            case .Red:
                funView.currentColor = UIColor.red
                funView.useRandomColor = false
            case .Blue:
                funView.currentColor = UIColor.blue
                funView.useRandomColor = false
            case .Yellow:
                funView.currentColor = UIColor.yellow
                funView.useRandomColor = false
            case .Green:
                funView.currentColor = UIColor.green
                funView.useRandomColor = false
            case .Random:
                funView.useRandomColor = true
            }
        }
    }
    
    @IBAction func changeShape(_ sender: UISegmentedControl) {
        let shapeSelection = Shape(rawValue: UInt(sender.selectedSegmentIndex))
        if let shape = shapeSelection {
            let funView = view as! QuartzFunView
            funView.shape = shape
            colorControl.isHidden = shape == Shape.Image
        }
    }
}

